app/controllers/queued_posts_controller.rb

```diff
@@ -20,6 +20,8 @@ def index
   def update
     qp = QueuedPost.where(id: params[:id]).first
 
+    return render_json_error "Post not found or already updated." if qp.blank?
```

Comment:
Tip here, any strings that are client facing needs to be translated. Also anytime you notice that you're duplicating a string in your application code and tests, it'll be much better to add the string to i18n so that we avoid duplicated strings.